scripts/check-format.sh

```diff
@@ -34,11 +34,11 @@ badly_named_files=""
 for file in $(find "$@" -name "*.h" -or -name "*.hpp" -or -name "*.cpp" -or -name "*.c"); do
   # Workaround for https://bugs.llvm.org/show_bug.cgi?id=39216
   d=$(cat "$file" | clang-format-8 -style=file --assume-filename "${file%.*}".cpp | diff "$file" -)
-  if $fix ; then
-    cat "$file" | clang-format-8 -style=file --assume-filename "${file%.*}".cpp > "$file".tmp
-    mv "$file".tmp "$file"
-  fi
   if [ "$d" != "" ]; then
+    if $fix ; then
+      cat "$file" | clang-format-8 -style=file --assume-filename "${file%.*}".cpp > "$file".tmp
```

Comment:
The bug entry on line 35 seems like it has been fixed in llvm 9, we should give that a try and see if we can't simplify this code quite a bit as a result.

Reply:
Another PR?

Reply:
Yes, absolutely.